src/lib/libast/tests/path/pathtemp.c

```diff
@@ -0,0 +1,28 @@
+#include "config_ast.h"  // IWYU pragma: keep
+
+#include <fcntl.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "ast.h"
+#include "terror.h"
+
+const char *temp_file;
+
+void cleanup_handler(void) { unlink(temp_file); }
```

Comment:
You're missing the `atexit()` call to install this handler. I'd also recommend initializing `temp_file` above to NULL and predicating the unlink on `if (temp_file)`.